cli/src/commands/rebase.rs

```diff
@@ -155,9 +155,9 @@ pub(crate) struct RebaseArgs {
     destination: Vec<RevisionArg>,
 
     /// If true, when rebasing would produce an empty commit, the commit is
-    /// skipped.
-    /// Will never skip merge commits with multiple non-empty parents.
-    /// Will never skip the working commit.
+    /// abandoned. It will not be abandoned if it was already empty before the
+    /// rebase. Will never skip merge commits with multiple non-empty
+    /// parents.
```

Comment:
Consider the following workflow, which I suspect to be relatively common:
1) `jj commit` (creating old_main -> my_commit (non-empty) -> @ (empty)
2) <upload and run CI. Realize you need to sync>
3) git fetch origin
4) `jj rebase -d `main`

IIUC, we would now have main -> my_commit (@)

This seems inconsistent with other commands (eg. `jj amend`) that abandon `@` then re-create `@`. Should we be doing that in this command too (maybe we already are, but it just needs to be documented here)?

If so, can we please update the documentation here to reflect that.

Reply:
> (maybe we already are, but it just needs to be documented here)?

Yes, we already do that.

After this commit, the working-copy commit is going to be preserved because it was already empty (and this commit change `--skip-empty` to use `AbandonNewlyEmpty`).

However, I think we had that behavior even before this commit because we always create a new, empty commit for the working copy the old working-copy commit is abandoned.

It's not documented here because it's not specific to `jj rebase`. I'm not sure where we should document it. Maybe we should actually document that behavior on (almost) every command because that's where people are going to see it? Anyway, I think it's orthogonal to this PR, so I'll leave it for a separate PR if you don't mind.

Reply:
Ah, LGTM then.

FWIW, my two cents is that almost every command should have it documented, because as you said, that's where people will see it. Otherwise we'll just end up with more bugs like that one filed internally the other day where amend confused them because the documentation suggested that it would make `@` non-empty.